### DISCUSSION TITLE
Documentation on how to write a test class when using triggers for Apex Rollup?

### DISCUSSION THREAD

**Discussion by baobao917** [jamessimone/apex-rollup#498] [Q&A]
I'm using apex triggers to invoke apex rollup.  I've gotten everything to work properly in a sandbox but I'm unable to deploy the apex triggers b/c I need test classes.  Sorry if this is a basic question for a developer (which I am not).  I don't see the documentation that walks us through this.

On another note, it would be really convenient if Apex Rollup had a feature like DLRS where you could click Manage child triggers and click Deploy and it would then create the triggers and test classes for you.

**Comment by jamessimone** [comment]
@baobao917 there is a plugin for Apex Rollup - check the plugins section of the README - that you can use to get code coverage for the package classes (if needed), but there's no plan to create Apex Triggers or classes automatically with Apex Rollup.

One of the biggest Achilles heel's of DLRS is its creation of triggers/classes. It creates scaling problems and leads to difficulty in debugging for most orgs. These issues were one of my primary inspirations in creating an alternative to DLRS to begin with!

My advice would be to use Flow to invoke Apex Rollup if you aren't able to write a small test class for the trigger you've written. I mean that in the best way possible; I know it won't be possible for every person to create test classes for their triggers, and that's precisely why the invocable versions of Apex Rollup exist. Hopefully this helps!

**Reply by baobao917** [reply]
Thanks for the advice/context.  I'll try that.
